server/datastore/mysql/hosts.go

```diff
@@ -744,27 +744,40 @@ func (ds *Datastore) CleanupIncomingHosts(ctx context.Context, now time.Time) ([
 	return ids, nil
 }
 
-func (ds *Datastore) GenerateHostStatusStatistics(ctx context.Context, filter fleet.TeamFilter, now time.Time, platform *string) (*fleet.HostSummary, error) {
+func (ds *Datastore) GenerateHostStatusStatistics(ctx context.Context, filter fleet.TeamFilter, now time.Time, platform *string, lowDiskSpace *int) (*fleet.HostSummary, error) {
 	// The logic in this function should remain synchronized with
 	// host.Status and CountHostsInTargets - that is, the intervals associated
 	// with each status must be the same.
 
 	args := []interface{}{now, now, now, now}
+	hostDisksJoin := ``
+	lowDiskSelect := `0 low_disk_space`
+	if lowDiskSpace != nil {
+		hostDisksJoin = `LEFT JOIN host_disks hd ON (h.id = hd.host_id)`
+		lowDiskSelect = `COALESCE(SUM(CASE WHEN hd.gigs_disk_space_available <= ? THEN 1 ELSE 0 END), 0) low_disk_space`
```

Comment:
Wait I'm a little confused, should this be using `percent_disk_space_available`? Isn't `lowDiskSpace` a number between 1 and 100? (https://github.com/fleetdm/fleet/pull/7872/files#diff-8103c94796fb91784bbc8c75c1fd026e47eee9db34bd5daf30b0aefcae460558R481) - also if we are going to use an Int type for filtering this, maybe we should store the data as an Int as well?

Reply:
Haha yeah it is confusing, we limit the `low_disk_space` to a number between 1-100 just so we don't blindly accept any value (that could result in a full disk scan), but this number is the number of GB, not a percentage (as per the feature request).

>  if we are going to use an Int type for filtering this, maybe we should store the data as an Int as well?

For the threshold parameter, we'll be using an integer as we won't start asking for hosts that have less than `10.25` GB or something, but the actual stored value (that you can still see I believe when navigating to the specific host's page) should still be as precise as what we request from osquery, and we request decimals.

Reply:
To clarify, for now the frontend will only ever pass `32` for the `low_disk_space` parameter, but we may add other thresholds in the future. For the API users, we (somewhat arbitrarily, but seems like a good starting point) accept 1-100.